File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        long[] fibonacci = new long[n+1];
        
        fibonacci[1]=1L;
        fibonacci[2]=1L;
        
        for(int i=2;i<=n;i++){
            fibonacci[i] = (fibonacci[i-1]+fibonacci[i-2])%1234567;
        }
        
        
        
        return (int)fibonacci[n];
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    answer = 0
    
    fibonacci = []
    fibonacci.append(1)
    fibonacci.append(1)    
    
    for i in range (2,n):
        fibonacci.append((fibonacci[i-1] + fibonacci[i-2]) % 1234567)
    
    return fibonacci[n-1]
